### PR TITLE
Update Connection.php for add +srv (mongodb+srv)

### PR DESCRIPTION
**Pull request by metola**
Fix issue #304
This is need if you create a cloud cluster https://cloud.mongodb.com/ and need conection mongodb+srv

| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | no
| Fixed issues  | #304

### CODE REVIEW

File: src/Connection.php [hunk 1]
```diff
@@ -200,8 +200,8 @@ public function setDefaultDatabaseName($name)
     public function getDefaultDatabaseName()
     {
         if ($this->_defaultDatabaseName === null) {
-            if (preg_match('/^mongodb:\\/\\/.+\\/([^?&]+)/s', $this->dsn, $matches)) {
-                $this->_defaultDatabaseName = $matches[1];
+            if (preg_match('/^mongodb(:|\+srv:)\\/\\/.+\\/([^?&]+)/s', $this->dsn, $matches)) {
+                $this->_defaultDatabaseName = $matches[2];
```
Comment (samdark) [inline src/Connection.php:204]:
```suggestion
            if (preg_match('/^mongodb(?:\+srv)?:\\/\\/.+\\/([^?&]+)/s', $this->dsn, $matches)) {
                $this->_defaultDatabaseName = $matches[1];
```

**Comment by samdark** [comment]
Also need a line for CHANGELOG.md

**Comment by ziaratban** [comment]
i think it is not a good way, because a variety  of mongodb connection string patterns exist.(Standalone & Replica Set & Shard).

the `defaultDatabaseName` parameter is required  for better performance and compatibility.

https://docs.mongodb.com/manual/reference/connection-string/

**Comment by metola** [comment]
Hi @ziaratban ,
what do you think is the best way to approach it?

**Comment by ziaratban** [comment]
hi  metola ,
I think the best solution is to not customize this part. this part must be set by programmer.

```php
return [
    'class' => 'yii\mongodb\Connection',
    'dsn' => 'mongodb://any syntax',
    'defaultDatabaseName' => 'myDB', // <<<< required 
    'enableLogging' => YII_ENV === 'dev',
    'enableProfiling' => YII_ENV === 'dev',
];
```

**Comment by metola** [comment]
Hi @ziaratban ,
Thanks, with this worlk well.

**Pull request closed without merging**